paseos/tests/attitude_test.py

```diff
@@ -0,0 +1,166 @@
+    """
+
+    earth = pk.planet.jpl_lp("earth")
+
+    # First actor constant angular acceleration
+    omega = np.pi / 2000
+
+    # Define first local actor with angular velocity
+    sat1 = ActorBuilder.get_actor_scaffold("sat1", SpacecraftActor, pk.epoch(0))
+    ActorBuilder.set_orbit(sat1, [7000000, 0, 0], [0, 8000.0, 0], pk.epoch(0), earth)
+    ActorBuilder.set_spacecraft_body_model(sat1, mass=100)
+    ActorBuilder.set_attitude_model(
+        sat1,
+        actor_initial_angular_velocity=[0.0, omega, 0.0],
+        actor_pointing_vector_body=[0, 0, 1],
+    )
+
+    # Define second local actor without angular velocity
+    sat2 = ActorBuilder.get_actor_scaffold("sat2", SpacecraftActor, pk.epoch(0))
+    ActorBuilder.set_orbit(sat2, [7000000, 0, 0], [0, 8000.0, 0], pk.epoch(0), earth)
+    ActorBuilder.set_spacecraft_body_model(sat2, mass=100)
+    ActorBuilder.set_attitude_model(
+        sat2,
+        actor_initial_angular_velocity=[0.0, 0.0, 0.0],
+        actor_pointing_vector_body=[0, 0, 1],
+    )
+
+    # Check Initial values
+
+    # sat1
+    assert np.all(sat1._attitude_model._actor_pointing_vector_body == [0.0, 0.0, 1.0])
+    assert np.all(sat1._attitude_model._actor_pointing_vector_eci == [-1.0, 0.0, 0.0])
+    assert np.all(sat1._attitude_model._actor_attitude_in_rad == [0.0, 0.0, 0.0])
+    # positive angular velocity in body y direction is negative angular velocity in Earth inertial z direction:
+    assert np.all(sat1._attitude_model._actor_angular_velocity_eci == [0.0, 0.0, -omega])
+
+    # sat2
+    assert np.all(sat2._attitude_model._actor_pointing_vector_body == [0.0, 0.0, 1.0])
+    assert np.all(sat2._attitude_model._actor_pointing_vector_eci == [-1.0, 0.0, 0.0])
+    assert np.all(sat2._attitude_model._actor_attitude_in_rad == [0.0, 0.0, 0.0])
+
+    # Initialise simulation
+    sim = paseos.init_sim(sat1)
+    sim.add_known_actor(sat2)
+    # Run simulation 20 steps
+    for i in range(20):
+        sim.advance_time(100, 0)
+
+    # Testing the simulation went as intended
+    # Pointing vector from sat1 must be rotated from [-1, 0, 0] to [1, 0, 0]:
+    assert np.all(np.isclose(sat1.pointing_vector, np.array([1.0, 0.0, 0.0])))
+    # Sat1 angular velocity in the body frame must stay constant:
+    assert np.all(
+        np.isclose(
+            sat1._attitude_model._actor_angular_velocity,
+            np.array([0.0, omega, 0.0]),
+        )
+    )
+    # Sat1 angular velocity in the Earth inertial frame must stay constant:
+    assert np.all(
+        np.isclose(
+            sat1.angular_velocity,
+            np.array([0.0, 0.0, -omega]),
+        )
+    )
+
+    # Pointing vector from sat2 must not be rotated.
+    assert np.all(sat2.pointing_vector == np.array([-1.0, 0.0, 0.0]))
+    # Sat2 angular velocity in the body frame must stay zero:
+    assert np.all(sat2._attitude_model._actor_angular_velocity == np.array([0.0, 0.0, 0.0]))
+
```

Comment:
only checking against trivial values, no actual check if the model produces correct values after e.g. 5 seconds of rotation